### ISSUE TITLE
False positive in immutable inspection

### ISSUE THREAD

**Issue by la10736** [intellij-rust/intellij-rust#1347]
Follow code yield a false immutable error:
```rust
struct I {
    parent: i32
}

struct P {
    v: Vec<I>
}

impl P {
    fn new() -> Self {
        P { v: vec![I { parent:0 }] }
    }

    fn mutate(&mut self, new_val: i32) {
        self.v[0].parent = new_val
      //^---------------------------------
    }
}

fn main() {
    let mut p = P::new();
    p.mutate(1);
    println!("{}", p.v[0].parent);
}
```

**Comment by la10736** [comment]
Both with 5 June and nightly release.

**Comment by matklad** [comment]
Closed by #1289!
